scan-command/src/main/java/io/ballerina/scan/utils/ScanUtils.java

```diff
@@ -276,6 +301,242 @@ private static void unzipReportResources(InputStream source, File target) throws
+                    }
+                }
+
+                if (Files.exists(Path.of(path))) {
+                    ScanTomlFile.Platform platform = new ScanTomlFile.Platform(name, path, properties);
+                    scanTomlFile.setPlatform(platform);
+                }
+            }
+        });
+
+        List<Toml> staticCodeAnalyzerTables = scanTomlDocumentContent.getTables(ANALYZER_TABLE);
+        staticCodeAnalyzerTables.forEach(staticCodeAnalyzer -> {
+            Map<String, Object> properties = staticCodeAnalyzer.toMap();
+            String org = !(properties.get("org") instanceof String) ? null :
+                    properties.get("org").toString();
+            String name = !(properties.get("name") instanceof String) ? null :
+                    properties.get("name").toString();
+
+            Pattern versionPattern = Pattern.compile(CUSTOM_RULES_COMPILER_PLUGIN_VERSION_PATTERN);
+            Object providedVersion = properties.get("version");
+            String version = (providedVersion instanceof String) &&
+                    versionPattern.matcher(providedVersion.toString()).matches() ? providedVersion.toString() : null;
+            String repository = !(properties.get("repository") instanceof String) ? null :
+                    properties.get("repository").toString();
+
+            if (org != null && !org.isEmpty() && name != null && !name.isEmpty()) {
+                ScanTomlFile.Analyzer analyzer;
+
+                if (repository != null && repository.equals(LOCAL_REPOSITORY_NAME) && version != null &&
+                        !version.isEmpty()) {
+                    analyzer = new ScanTomlFile.Analyzer(org, name, version, repository);
+                } else {
+                    analyzer = new ScanTomlFile.Analyzer(org, name, version, null);
+                }
+                scanTomlFile.setAnalyzer(analyzer);
+            }
+        });
+
+        Toml rulesTable = scanTomlDocumentContent.getTable(RULES_TABLE).orElse(null);
+        if (rulesTable != null) {
+            TomlValueNode include = rulesTable.get("include").orElse(null);
+            TomlArrayValueNode includeArray;
+            if (include != null && include.kind().equals(TomlType.ARRAY)) {
+                includeArray = (TomlArrayValueNode) include;
+                List<String> rulesToInclude = includeArray.toNativeValue()
+                        .stream()
+                        .map(Object::toString)
+                        .toList();
+
+                rulesToInclude.forEach(ruleToInclude -> {
+                    ScanTomlFile.RuleToFilter ruleToFilter = new ScanTomlFile.RuleToFilter(ruleToInclude);
+                    scanTomlFile.setRuleToInclude(ruleToFilter);
+                });
+            }
+
+            TomlValueNode exclude = rulesTable.get("exclude").orElse(null);
+            TomlArrayValueNode excludeArray = null;
+            if (exclude != null && exclude.kind().equals(TomlType.ARRAY)) {
```

Comment:
Same as `include`, Also since the inner logic seems similar unless it is accidental may be factor it to a separate method